src/keystore/persistence/EncryptedPersistence.ts

```diff
@@ -0,0 +1,82 @@
+import { Persistence } from './interface'
+import eccrypto, { Ecies } from 'eccrypto'
+
+const assertEciesLengths = (ecies: Ecies): void => {
+  if (ecies.iv.length !== 16) {
+    throw new Error('Invalid iv length')
+  }
+  if (ecies.ephemPublicKey.length !== 65) {
+    throw new Error('Invalid ephemPublicKey length')
+  }
+  if (ecies.ciphertext.length < 1) {
+    throw new Error('Invalid ciphertext length')
+  }
+  if (ecies.mac.length !== 32) {
+    throw new Error('Invalid mac length')
+  }
+}
+
+const serializeEcies = (ecies: Ecies): Uint8Array => {
+  assertEciesLengths(ecies)
+  const { iv, ephemPublicKey, ciphertext, mac } = ecies
+  const result = new Uint8Array(
+    iv.length + ephemPublicKey.length + ciphertext.length + mac.length
+  )
+  result.set(iv, 0)
+  result.set(ephemPublicKey, iv.length)
+  result.set(ciphertext, iv.length + ephemPublicKey.length)
+  result.set(mac, iv.length + ephemPublicKey.length + ciphertext.length)
+  return result
+}
+
+const deserializeEcies = (data: Uint8Array): Ecies => {
```

Comment:
This looks dangerous. 
the length of data is not verified prior to usage. Input bytes should be checked to ensure they are a sane length.

Reply:
What would a sane length be? > 132?

Reply:
For the serialization above I get:

```
fixed_length = IV_LENGTH + EPHEMERAL_PUBLIC_KEY_LENGTH + MAC_LENGTH + AES_BLOCKSIZE //(minimum 1 block of ciphertext) = 129

// Ensure data is atleast 'fixed_length' bytes and the remaining bytes are valid AES_blocks
(data.length > fixed_length) && ((data.length - fixed_length) % AES_BLOCKSIZE) == 0
```

Reply:
Updated